Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/LoadTestDataPatientBean.java

```diff
@@ -83,82 +113,246 @@ public void setSelectedPatient(Patient selectedPatient) {
         this.selectedPatient = selectedPatient;
     }
 
-    // DATABASE-METHODS-getAll, delete, save
+    // selected-record
+    public Personname getSelectedPersonname() {
+        return selectedPersonname;
+    }
+
+    public void setSelectedPersonname(Personname selectedPersonname) {
+        this.selectedPersonname = selectedPersonname;
+    }
+
+    public Identifier getSelectedIdentifier() {
+        return selectedIdentifier;
+    }
+
+    public void setSelectedIdentifier(Identifier selectedIdentifier) {
+        this.selectedIdentifier = selectedIdentifier;
+    }
+
+    public Address getSelectedAddress() {
+        return selectedAddress;
+    }
+
+    public void setSelectedAddress(Address selectedAddress) {
+        this.selectedAddress = selectedAddress;
+    }
+
+    public Phonenumber getSelectedPhonenumber() {
+        return selectedPhonenumber;
+    }
+
+    public void setSelectedPhonenumber(Phonenumber selectedPhonenumber) {
+        this.selectedPhonenumber = selectedPhonenumber;
+    }
+
+    // database-methods
+    public List<Phonenumber> getPhonenumbers() {
+        if (isValidPatientId()) {
+            return loadTestDataService.getAllPhonenumbersBy(getPatientId());
+        } else {
+            return new ArrayList<>();
+        }
+    }
+
+    public boolean deletePhonenumber() {
+        if (selectedPhonenumber != null) {
+            return loadTestDataService.deletePhonenumber(selectedPhonenumber);
+        } else {
+            addPatientErrorMessages("Phonenumber have to be select to be deleted");
+        }
+        return false;
+    }
+
+    public boolean savePhonenumber() {
+        boolean actionResult = false;
```

Comment:
1.  Refactor the save code GenericDAO to remove the redundant logic.


2. To be consistent with other DML operations in GenericDAOImpl.java, initialize the actionResult to true and change the logic accordingly. Refer GenericDAOImpl.update method.

Reply:
don't think we wanted to bring the DAO into the bean layer